tests/lib/devicetree/devices fails to build for thingy52_nrf52832

**Describe the bug**
An attempt to build `tests/lib/devicetree/devices` for `thingy52_nrf52832` ends up with the following linking error:
```
arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/regulator/libdrivers__regulator.a(regulator_fixed.c.obj):(.rodata.regulator_2_cfg+0xc): undefined reference to `__device_dts_ord_8'
arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/regulator/libdrivers__regulator.a(regulator_fixed.c.obj):(.rodata.regulator_3_cfg+0xc): undefined reference to `__device_dts_ord_8'
arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/regulator/libdrivers__regulator.a(regulator_fixed.c.obj):(.rodata.regulator_4_cfg+0xc): undefined reference to `__device_dts_ord_8'
```
This happens because this board needs the SX1509B GPIO expander driver for the regulator devices it uses and this driver depends on I2C which needs to be disabled for this test.

This is the same kind of problem that has been addressed by commit https://github.com/zephyrproject-rtos/zephyr/commit/b579c12714a452ddcb436615d218d39f9f0cd5bd for hsdk platforms.

**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -DBOARD=thingy52_nrf52832 ..
3. make
4. See error

**Impact**
CI fails for certain PRs, for example: https://github.com/zephyrproject-rtos/zephyr/pull/40852.

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.13.2
 - Commit f92542ecf3e465861ae63fc05932fb229b527801

--- a/tests/lib/devicetree/devices/testcase.yaml
+++ b/tests/lib/devicetree/devices/testcase.yaml
@@ -5,4 +5,4 @@ tests:
     # will mostly likely be the fastest.
     integration_platforms:
       - native_posix
-    platform_exclude: hsdk hsdk_2cores
+    platform_exclude: hsdk hsdk_2cores thingy52_nrf52832

